proto/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# This Dockerfile is used for proto generation
+# To build, run `make proto-image-build`
+
+FROM bufbuild/buf:1.7.0 as BUILDER
```

Comment:
Do we have some simple docs or a README on how to generate the protos as well as for which context we use them? If not, let's please add them

Reply:
That is a good point, I was thinking adding them in later stages for some reason, but since we already introduce some proto files with this PR it makes sense to include them right now. I have added relevant make files and scripts that will do do job when calling our existing `proto-gen` command.